src/legendre.rs

```diff
@@ -0,0 +1,35 @@
+use ff::{Field, PrimeField};
+
+pub trait Legendre: Field {
+    type BasePrimeField: PrimeField;
+
+    // This is (p-1)/2 where p is the modulus of the base prime field
+    fn legendre_exp() -> &'static Vec<u64>;
+
+    fn norm(&self) -> &Self::BasePrimeField;
```

Comment:
Is this added to cover the extension field? If so I think we couldn't return a reference since the `norm` of `Fq2` is calculated on the fly, so here we'd have some lifetime issue.